application/src/test/java/md/maib/retail/services/FindCampaignByIdServiceTest.java

```diff
@@ -0,0 +1,45 @@
+package md.maib.retail.services;
+
+import md.maib.retail.CampaignAllInfo;
+import md.maib.retail.find_campaign_by_id.FindCampaignByIdService;
+import md.maib.retail.find_campaign_by_id.FindCampaignByIdUseCase;
+import md.maib.retail.junit.UnitTest;
+import md.maib.retail.model.campaign.Campaign;
+import md.maib.retail.model.campaign.CampaignId;
+import md.maib.retail.model.campaign.CampaignState;
+import md.maib.retail.model.ports.Campaigns;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@UnitTest
+public class FindCampaignByIdServiceTest {
+    Campaigns campaigns;
```

Comment:
How you mock this port?

may be use org.mockito.Mockito?

annotation @Mock on Campaigns campaigns

special methods
@BeforeEach
    void setupProcessor() {
        target = findCampaignsByIdService(campaigns);
    }

    @AfterEach
    void tearDown() {
        verifyNoMoreInteractions(campaigns);
    }

and in test methods something like that
when(campaigns.findActive(any())).thenReturn(Optional.of(campaign));

Reply:
done